### PR TITLE
	Support Quick Login from Genius Center

### PR DESCRIPTION
**Pull request by acearth**

### CODE REVIEW

File: app/components/Signin/index.jsx [hunk 1]
```diff
@@ -26,10 +26,16 @@ var Login = React.createClass({
         }
     },
     componentWillMount(){
-        this.props.setTitle("RhinoBird");
-    },
-    componentDidMount() {
-        this.refs.email.focus();
+        var query = this.context.router.getCurrentQuery();
+        console.log(query);
```
Comment (WangBoxue) [inline app/components/Signin/index.jsx]: remove this debug info

File: app/components/Signin/index.jsx [hunk 2]
```diff
@@ -26,10 +26,16 @@ var Login = React.createClass({
...
+        if (query.ticket) {
```
Comment (WangBoxue) [inline app/components/Signin/index.jsx]: What if query is null or undefined?

File: app/components/Signin/index.jsx [hunk 3]
```diff
@@ -26,10 +26,16 @@ var Login = React.createClass({
...
+            $.post('/platform/api/genius_coming', {ticket: query.ticket, sign: query.sign}).then((data) => {
+                LoginAction.updateLogin(data);
+                this.context.router.transitionTo(this.context.router.getCurrentQuery().target || "/");
+            });
+        } else {
+            this.props.setTitle("RhinoBird");
```
Comment (WangBoxue) [inline app/components/Signin/index.jsx:36]: We should always set title

Comment (acearth) [inline app/components/Signin/index.jsx]: Even when the URL doesn't  contain any query, the query object is not NULL.  So the usage of 'query.ticket' is safe.    

If ticket parameter exists, it's means this request is coming from Genius Center's callback. So the page will be redirect to User page, so set title in this page is meanless.

Comment (acearth) [reply app/components/Signin/index.jsx:36]: Even when the URL doesn't contain any query, the query object is not NULL. So the usage of 'query.ticket' is safe.

If ticket parameter exists, it's means this request is coming from Genius Center's callback. So the page will be redirect to User page, so set title in this page is meanless.

File: app/components/Signin/index.jsx [hunk 4]
```diff
@@ -26,10 +26,15 @@ var Login = React.createClass({
         }
     },
     componentWillMount(){
-        this.props.setTitle("RhinoBird");
-    },
-    componentDidMount() {
-        this.refs.email.focus();
```
Comment (acearth) [inline app/components/Signin/index.jsx:32]: User will be redirect to service provider's(SP) callback_URL(set by SP admin when they register SP) after successfully logged in Genius Center.  ticket parameter will be attached to the callback_URL.
When SP received the callback request, SP should send request to Genius center to fetch user's information.
In this case(Rhinobird),  the request is proxied by backend server.

File: app/components/Signin/index.jsx [hunk 5]
```diff
@@ -46,17 +51,27 @@ var Login = React.createClass({
             });
         });
     },
+
+    _quick_login(){
```
Comment (acearth) [inline app/components/Signin/index.jsx:55]: Login Page redirected to Genius Center. Please remember add "app_id" parameter(IMPORTANT).

**Pull request merged**